magicctapipe/scripts/lst1_magic/setting_up_config_and_dir.py

```diff
@@ -0,0 +1,466 @@
+    """
+    
+    f = open(target_dir+'/config_DL0_to_DL1.yaml','w')
+    #f.write("directories:\n    target: "+target_dir+"\n\n")    
+    lines_of_config_file = [
+    "mc_tel_ids:",
+    "\n    LST-1: "+str(ids[0]),
+    "\n    LST-2: "+str(ids[1]),
+    "\n    LST-3: "+str(ids[2]),
+    "\n    LST-4: "+str(ids[3]),
+    "\n    MAGIC-I: "+str(ids[4]),
+    "\n    MAGIC-II: "+str(ids[5]),
+    "\n",
+    "\nLST:",
+    "\n    image_extractor:",
+    '\n        type: "LocalPeakWindowSum"',
+    "\n        window_shift: 4",
+    "\n        window_width: 8",
+    "\n",
+    "\n    increase_nsb:",
+    "\n        use: true",
+    "\n        extra_noise_in_dim_pixels: 1.27",
+    "\n        extra_bias_in_dim_pixels: 0.665",
+    "\n        transition_charge: 8",
+    "\n        extra_noise_in_bright_pixels: 2.08",
+    "\n",
+    "\n    increase_psf:",
+    "\n        use: false",
+    "\n        fraction: null",
+    "\n",
+    "\n    tailcuts_clean:",
+    "\n        picture_thresh: 8",
+    "\n        boundary_thresh: 4",
+    "\n        keep_isolated_pixels: false",
+    "\n        min_number_picture_neighbors: 2",
+    "\n",
+    "\n    time_delta_cleaning:",
+    "\n        use: true",
+    "\n        min_number_neighbors: 1",
+    "\n        time_limit: 2",
+    "\n",
+    "\n    dynamic_cleaning:",
+    "\n        use: true",
+    "\n        threshold: 267",
+    "\n        fraction: 0.03",
+    "\n",
+    "\n    use_only_main_island: false",
+    "\n",
+    "\nMAGIC:",
+    "\n    image_extractor:",
+    '\n        type: "SlidingWindowMaxSum"',
+    "\n        window_width: 5",
+    "\n        apply_integration_correction: false",
+    "\n",
+    "\n    charge_correction:",
+    "\n        use: true",
+    "\n        factor: 1.143",
+    "\n",
+    "\n    magic_clean:",
+    "\n        use_time: true",
+    "\n        use_sum: true",
+    "\n        picture_thresh: 6",
+    "\n        boundary_thresh: 3.5",
+    "\n        max_time_off: 4.5",
+    "\n        max_time_diff: 1.5",
+    "\n        find_hotpixels: true",
+    '\n        pedestal_type: "from_extractor_rndm"',
+    "\n",
+    "\n    muon_ring:",
+    "\n        thr_low: 25",
+    "\n        tailcut: [12, 8]",
+    "\n        ring_completeness_threshold: 25",
+    "\n"]
+    
+    f.writelines(lines_of_config_file)
+    f.close()
+
+
+def lists_and_bash_generator(particle_type, target_dir, MC_path, SimTel_version, focal_length, scripts_dir):
+
+    """
+    This function creates the lists list_nodes_gamma_complete.txt and list_folder_gamma.txt with the MC file paths.
+    After that, it generates a few bash scripts to link the MC paths to each subdirectory. 
+    These bash scripts will be called later in the main() function below. 
+    """
+    
+    process_name = target_dir.split("/")[-2:][1]
+    
+    list_of_nodes = glob.glob(MC_path+"/node*")
+    f = open(target_dir+f"/list_nodes_{particle_type}_complete.txt","w") # creating list_nodes_gammas_complete.txt
```

Comment:
better to use the `with open(....) as f:` construct so you do not need to call `f.close()`, since python will do it for you when `f` goes out of scope

Reply:
Should be all fixed now